src/app/model-download/components/model-download-component/model-download-component.component.html

```diff
@@ -0,0 +1,231 @@
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                                Revision</th>
+                            <th scope="col"
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                                Link</th>
+                            <th scope="col"
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                                Download date</th>
+                            <th scope="col"
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                                Size</th>
+                            <th scope="col"
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                                Type</th>
+                            <th scope="col"
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                                Status</th>
+                            <th scope="col"
+                                class="px-3 py-2 text-center text-xs font-medium uppercase tracking-wide text-gray-500">
+                            </th>
+                        </tr>
+                    </thead>
+                    <tbody class="divide-y divide-gray-200">
+                        <tr *ngFor="let model of downloadedModels; let i = index">
+                            <td class="text-center px-3 py-2 text-sm text-gray-500">{{model.name}}</td>
+                            <td class="text-center px-3 py-2 text-sm text-gray-500">
+                                <ng-template [ngIf]="model.revision">{{model.revision}}</ng-template>
+                            </td>
+                            <td class="text-center px-3 py-2 text-sm text-gray-500">
+                                <a [href]="model.link" target="_blank" *ngIf="model.link" class="h4 w-4 m-auto block"
+                                    style="padding:0px"><svg xmlns="http://www.w3.org/2000/svg" class="h-4 w-4 m-auto"
+                                    fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+                                    <path stroke-linecap="round" stroke-linejoin="round"
+                                        d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14" />
+                                </svg></a>
+                            </td>
+                            <td class="whitespace-nowrap text-center px-3 py-2 text-sm text-gray-500">
+                                <ng-template [ngIf]="model.date != 0 && (model.status === 'finished' || model.status === 'downloading')">{{parseUTC(model.date)}}</ng-template>
```

Comment:
functions in html code are run quite often. changing the underlying donwloadedModels variable to be fully prepared on subscription would reduce workload (new field `parsedDate` or something)